src/cpp/rtps/participant/RTPSParticipantImpl.cpp

```diff
@@ -449,24 +449,50 @@ RTPSParticipantImpl::~RTPSParticipantImpl()
 
 template <EndpointKind_t kind, octet no_key, octet with_key>
 bool RTPSParticipantImpl::preprocess_endpoint_attributes(
-        const char* debug_label,
         const EntityId_t& entity_id,
         EndpointAttributes& att,
         EntityId_t& entId)
 {
     if (!att.unicastLocatorList.isValid())
     {
-        logError(RTPS_PARTICIPANT, "Unicast Locator List for " << debug_label << " contains invalid Locator");
+        if (att.endpointKind == WRITER)
```

Comment:
Leave each of the logs as they were and set up the `debug_label` variable at the beginning of the function. This way you avoid duplicating code. I suggest you use the ternary operator for better code readability.

Reply:
:+1: